cmdLine/src/main/scala/topgun/cmdline/FileParser.scala

```diff
@@ -1,39 +1,44 @@
 package topgun.cmdline
 
 import java.io.File
-import java.nio.file.Files
-
-import com.jrockit.mc.common.IMCFrame
-import com.jrockit.mc.flightrecorder.FlightRecordingLoader
-import com.jrockit.mc.flightrecorder.internal.model.{FLRStackTrace, FLRThread, FLRType}
-import com.jrockit.mc.flightrecorder.spi.IEvent
+import java.util.stream.Collectors
+import java.util
+import jdk.jfr.consumer.{RecordedClass, RecordedEvent, RecordedFrame, RecordingFile}
 import topgun.core.CallSite
 
 import scala.jdk.CollectionConverters._
 
 class FileParser(file: File, cmdLine: JfrParseCommandLine, totals: Totals, configuration: Configuration) {
 
+  var allocationInNewTlab: Boolean = false
+  var allocationOutsideTlab: Boolean = false
+  var methodProfilingSample: Boolean = false
+
   import configuration.{includeStack, includeThread}
 
+  def foundRequiredEvents(): Boolean = {
```

Comment:
please move this method into `def parse`

Reply:
Okay Mike

Reply:
done